guides/05_chatbots/03_agents-and-tool-usage.md

```diff
@@ -166,4 +167,139 @@ demo.launch()
 That's it! See our finished langchain demo [here](https://huggingface.co/spaces/gradio/langchain-agent).
 
 
+## Building with Visible Thinking LLMs
```

Comment:
```suggestion
## Chain-of-Thought (CoT) demos
```

Reply:
I believe the terms 'Thinking' and 'Reasoning' will be increasingly popular for LLMS. Therefore, I want to ensure that our messaging is aligned. In my opinion, 'thinking' is currently more commonly used than 'Chain-of-thought.' In fact, Google has referred to its model as 'Gemini2.0 Flash Thinking Experimental,' so adopting 'Thinking' seems more logical to me.